File: minnkyoung/프린터.py
def solution(priorities, location):
    loc = [i for i in range(len(priorities))]
    final_loc = []

    while len(priorities) != 0:
        if priorities[0] == max(priorities):
            final_loc.append(loc.pop(0))
            priorities.pop(0)
        else:
            priorities.append(priorities.pop(0))
            loc.append(loc.pop(0))

    return final_loc.index(location) + 1

File: ayeon/프린터.cpp
/* SOLVED */

#include <string>
#include <vector>
#include <queue>

using namespace std;

typedef struct{
	int x; //index
	int y; //priority
}q;


int solution(vector<int> priorities, int location) {
    
    int count = 0;
    queue<q> Q;
    priority_queue<int, vector<int>> heap;
    
    for(int i=0; i<priorities.size(); i++){
        Q.push({i,priorities[i]});
        heap.push(priorities[i]);
    }
    
    while(!Q.empty())
    {
        if (Q.front().y < heap.top())
        {
            	Q.push(Q.front());
	            Q.pop();
        }
        else
        {
            count++;
            if (Q.front().x == location) return count;
            Q.pop();
            heap.pop();
        }           
    }
    return count;
}
